client-v2/src/components/util/EditModeVisibility.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { Fragment, ReactNode } from 'react';
+import { EditMode } from 'types/EditMode';
+import { selectEditMode } from 'selectors/pathSelectors';
+import { State } from 'types/State';
+import { connect } from 'react-redux';
+
+interface EditModeVisibilityProps {
+  currentMode: EditMode;
+  visibleMode: EditMode;
+  children: ReactNode;
+}
+
+const EditModeVisibility = ({
```

Comment:
This component feels useful for the feature switching work @jonathonherbert and @jennygrahamjones are doing for inline editing.